python/mach_bootstrap.py

```diff
@@ -154,19 +148,21 @@ def wptserve_path(is_firefox, topdir, *paths):
 def _activate_virtualenv(topdir, is_firefox):
     virtualenv_path = os.path.join(topdir, "python", "_virtualenv")
     check_exec_path = lambda path: path.startswith(virtualenv_path)
-    python = _get_exec_path(PYTHON_NAMES)   # If there was no python, mach wouldn't have run at all!
+    python = sys.executable   # If there was no python, mach wouldn't have run at all!
     if not python:
         sys.exit('Failed to find python executable for starting virtualenv.')
 
     script_dir = _get_virtualenv_script_dir()
     activate_path = os.path.join(virtualenv_path, script_dir, "activate_this.py")
     need_pip_upgrade = False
     if not (os.path.exists(virtualenv_path) and os.path.exists(activate_path)):
-        virtualenv = _get_exec_path(VIRTUALENV_NAMES)
-        if not virtualenv:
+        import imp
+        try:
+            imp.find_module('virtualenv')
```

Comment:
This change broke our Linux CI for some reason :\

Reply:
The only thing I can think of why it'd work before but not now is that virtualenv is available via binary, but not via the python which is being used to execute the bootstrap? 

The problem is that if you assume the environment, it can grab entirely the wrong python for specific steps and create the wrong _virtualenv silently. This happened to me while trying to get mach working on Windows (where `python` is 3.7).